boards/nRF52840-DK/src/lib.rs

```diff
@@ -0,0 +1,580 @@
+
+    /// nRF52 peripheral: EGU5
+    pub EGU5: nrf52::EGU5,
+
+    /// nRF52 peripheral: TIMER3
+    pub TIMER3: nrf52::TIMER3,
+
+    /// nRF52 peripheral: TIMER4
+    pub TIMER4: nrf52::TIMER4,
+
+    /// nRF52 peripheral: PWM0
+    pub PWM0: nrf52::PWM0,
+
+    /// nRF52 peripheral: PDM
+    pub PDM: nrf52::PDM,
+
+    /// nRF52 peripheral: NVMC
+    pub NVMC: nrf52::NVMC,
+
+    /// nRF52 peripheral: PPI
+    pub PPI: nrf52::PPI,
+
+    /// nRF52 peripheral: MWU
+    pub MWU: nrf52::MWU,
+
+    /// nRF52 peripheral: PWM1
+    pub PWM1: nrf52::PWM1,
+
+    /// nRF52 peripheral: PWM2
+    pub PWM2: nrf52::PWM2,
+
+    /// nRF52 peripheral: RTC2
+    pub RTC2: nrf52::RTC2,
+
+    /// nRF52 peripheral: I2S
+    pub I2S: nrf52::I2S,
+}
+
+impl nRF52840DK {
+    /// Take the peripherals safely
+    ///
+    /// This method will return an instance of `nRF52840DK` the first time it is
+    /// called. It will return only `None` on subsequent calls.
+    pub fn take() -> Option<Self> {
+        Some(Self::new(
+            CorePeripherals::take()?,
+            Peripherals::take()?,
+        ))
+    }
+
+    /// Steal the peripherals
+    ///
+    /// This method produces an instance of `nRF52840DK`, regardless of whether
+    /// another instance was create previously.
+    ///
+    /// # Safety
+    ///
+    /// This method can be used to create multiple instances of `nRF52840DK`. Those
+    /// instances can interfere with each other, causing all kinds of unexpected
+    /// behavior and circumventing safety guarantees in many ways.
+    ///
+    /// Always use `nRF52840DK::take`, unless you really know what you're doing.
+    pub unsafe fn steal() -> Self {
+        Self::new(
+            CorePeripherals::steal(),
+            Peripherals::steal(),
+        )
+    }
+
+    fn new(cp: CorePeripherals, p: Peripherals) -> Self {
+        let pins0 = p.P0.split();
+        let pins1 = p.P1.split();
+
+        // The nRF52840-DK has an 64MB SPI flash on board which can be interfaced through SPI or Quad SPI.
+        // As for now, only the normal SPI mode is available, so we are using this for the interface.
+        let flash_spim = p.SPIM2.constrain(spim::Pins {
+            sck : pins0.p0_19.into_push_pull_output(Level::Low).degrade(),
+            mosi: pins0.p0_20.into_push_pull_output(Level::Low).degrade(),
+            miso: pins0.p0_21.into_floating_input().degrade(),
+        });
+
+        let flash_cs = pins0.p0_17.into_push_pull_output(Level::High).degrade();
+        // let flash_wp = pins.p0_22.into_push_pull_output(Level::High).degrade();
+        // let flash_hold = pins.p0_23.into_push_pull_output(Level::High).degrade();
+
+        // The nRF52840-DK features a virtual COM port.
+        // It features HWFC but does not have to use it.
+        // It can transmit a flexible baudrate of up to 1Mbps.
+        let com_uart = p.UARTE0.constrain(uarte::Pins {
+                txd: pins0.p0_06.into_push_pull_output(Level::High).degrade(),
```

Comment:
This will need to change when #43 lands. Also dead code for cts/rts below

Reply:
Yes, I didn't change it yet as I have to wait for #43. No point in fixing it as well ;)
I reenabled HWFC as the CDC UART can do HWFC and it was eliminated as a potential source of error for the broken UART :)